docs/src/pages/customization/components/GlobalThemeVariants.tsx

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import {
+  createMuiTheme,
+  makeStyles,
+  Theme,
+  ThemeProvider,
+} from '@material-ui/core/styles';
+import Button from '@material-ui/core/Button';
+
+declare module '@material-ui/core/Button/Button' {
+  interface ButtonPropsVariantOverrides {
+    dashed: true;
+  }
+}
+
+const useStyles = makeStyles((theme: Theme) => ({
+  root: {
+    '& > *': {
+      margin: theme.spacing(1),
+    },
+  },
+}));
+
+const inputTheme = (outerTheme: Theme) =>
+  createMuiTheme({
+    variants: {
+      MuiButton: [
+        {
+          matcher: { variant: 'dashed' },
+          styles: {
+            padding: '5px 15px',
+            border: `5px dashed ${outerTheme.palette.primary.main}`,
+          },
+        },
+        {
+          matcher: { variant: 'dashed', color: 'secondary' },
+          styles: {
+            border: `5px dashed ${outerTheme.palette.secondary.main}`,
+          },
+        },
+      ],
+    },
+  });
+
+export default function GlobalThemeVariants() {
+  const classes = useStyles();
+
+  return (
+    <ThemeProvider theme={inputTheme}>
+      <div className={classes.root}>
```

Comment:
I would recommend this diff:

```diff
diff --git a/docs/src/pages/customization/components/GlobalThemeVariants.tsx b/docs/src/pages/customization/components/GlobalThemeVariants.tsx
index 060afadf9d..628762874b 100644
--- a/docs/src/pages/customization/components/GlobalThemeVariants.tsx
+++ b/docs/src/pages/customization/components/GlobalThemeVariants.tsx
@@ -21,38 +21,39 @@ const useStyles = makeStyles((theme: Theme) => ({
   },
 }));

-const inputTheme = (outerTheme: Theme) =>
-  createMuiTheme({
-    variants: {
-      MuiButton: [
-        {
-          matcher: { variant: 'dashed' },
-          styles: {
-            padding: '5px 15px',
-            border: `5px dashed ${outerTheme.palette.primary.main}`,
-          },
+const defaultTheme = createMuiTheme();
+
+const theme = createMuiTheme({
+  variants: {
+    MuiButton: [
+      {
+        matcher: { variant: 'dashed' },
+        styles: {
+          padding: '5px 15px',
+          border: `5px dashed ${defaultTheme.palette.primary.main}`,
         },
-        {
-          matcher: { variant: 'dashed', color: 'secondary' },
-          styles: {
-            border: `5px dashed ${outerTheme.palette.secondary.main}`,
-          },
+      },
+      {
+        matcher: { variant: 'dashed', color: 'secondary' },
+        styles: {
+          border: `5px dashed ${defaultTheme.palette.secondary.main}`,
         },
-      ],
-    },
-  });
+      },
+    ],
+  },
+});

 export default function GlobalThemeVariants() {
   const classes = useStyles();

   return (
-    <ThemeProvider theme={inputTheme}>
-      <div className={classes.root}>
+    <div className={classes.root}>
+      <ThemeProvider theme={theme}>
         <Button variant="dashed">Dashed</Button>
         <Button variant="dashed" color="secondary">
           Dashed secondary
         </Button>
-      </div>
-    </ThemeProvider>
+      </ThemeProvider>
+    </div>
   );
 }
```

so the demo works in codesandbox and to reduce the size of the inline preview from 

<img width="843" alt="Capture d’écran 2020-07-12 à 23 12 26" src="https://user-images.githubusercontent.com/3165635/87256746-a5db3800-c495-11ea-82ce-a24bd96f19a9.png">

to

<img width="840" alt="Capture d’écran 2020-07-12 à 23 13 04" src="https://user-images.githubusercontent.com/3165635/87256749-a7a4fb80-c495-11ea-8eba-840364332e48.png">